src/sp/it/pl/gui/objects/contextmenu/ImprovedContextMenu.kt

```diff
@@ -103,34 +110,53 @@ class ContextMenuItemSuppliers {
     private val mMany = ClassListMap<ItemsSupply> { fail() }
 
     @Suppress("UNCHECKED_CAST")
-    fun <T: Any> add(type: Class<T>, items: (ImprovedContextMenu<*>, T) -> Sequence<MenuItem>) {
-        mSingle.accumulate(type, items as ItemsSupply)
+    fun <T: Any> add(type: Class<T>, items: ContextMenuBuilder<T>.() -> Unit) {
+        mSingle.accumulate(type) { menu, item ->
+            ContextMenuBuilder(menu, item as T).also { items(it) }.asSequence()
+        }
     }
 
     @Suppress("UNCHECKED_CAST")
-    fun <T: Any> addMany(type: Class<T>, items: (ImprovedContextMenu<*>, Collection<T>) -> Sequence<MenuItem>) {
-        mMany.accumulate(type, items as ItemsSupply)
+    fun <T: Any> addMany(type: Class<T>, items: ContextMenuBuilder<Collection<T>>.() -> Unit) {
```

Comment:
This should now be inlined.

Reply:
Except it can not due to non local returns hmmm. I see I see. That makes sense but still I figure there must be some way...

Reply:
Nope:
Public-API inline function cannot access non-public-API 'private final val mMany: ClassListMap<ItemsSupply /* = (ImprovedContextMenu<*>, Any?) -> Sequence<MenuItem> */>

This can't and shouldn't be inlined

Reply:
Yes I already wrote that in my previous reply. But conceptually we do want to inline, because the lambda we use is theoretically inlinable, but because it is used in another lambda which is not things get iffy. No way around I suppose.

Reply:
Ah your problem with private api ca be fixed by additional methods that hide it. But non local return is the real problem if you go that route.

Reply:
I see. I only now realized that crossinline and noinline are not the same. In that case we may want to use crossinline. It will require useless methods to hide the internals though huh? Perhaps it is good as it is. What do you think?